modules/ref-inference-endpoints.adoc

```diff
@@ -0,0 +1,220 @@
+'{"requests": [{"text":"<text>"}]}' \
+-insecure <inference_endpoint_url>:443 fmaas.GenerationService/Generate \
+-H 'Authorization: Bearer <token>'
+----
+
+*OpenVINO Model Server*
+[source]
+----
+curl -ks <inference_endpoint_url>/v2/models/<model_name>/infer -d \
+'{ "model_name": "<model_name>", \
+"inputs": [{ "name": "<name_of_model_input>", "shape": [<shape>], "datatype": "<data_type>", "data": [<data>] }]}' \
+-H 'Authorization: Bearer <token>'
+----
+
+*vLLM ServingRuntime for KServe*
+[source]
+----
+curl -v https://<inference_endpoint_url>:443/v1/chat/completions -H \
+"Content-Type: application/json" -d '{ \
+"messages": [{ \
+"role": "<role>", \
+"content": "<content>" \
+}] -H 'Authorization: Bearer <token>'
+----
+--
+endif::[]
+ifdef::self-managed,cloud-service[]
+
++
+--
+*Caikit TGIS ServingRuntime for KServe*
+[source]
+----
+curl --json '{"model_id": "<model_name__>", "inputs": "<text>"}' https://<inference_endpoint_url>:443/api/v1/task/server-streaming-text-generation -H 'Authorization: Bearer <token>'
+----
+
+*Caikit Standalone ServingRuntime for KServe*
+
+.REST
+[source]
+----
+curl -H 'Content-Type: application/json' -d '{"inputs": "<text>", "model_id": "<model_id>"}' <inference_endpoint_url>/api/v1/task/embedding -H 'Authorization: Bearer <token>'
+----
+
+.gRPC
+[source]
+----
+grpcurl -insecure -d '{"text": "<text>"}' -H \"mm-model-id: <model_id>\" <inference_endpoint_url>:443 caikit.runtime.Nlp.NlpService/EmbeddingTaskPredict -H 'Authorization: Bearer <token>'
+----
+
+*TGIS Standalone ServingRuntime for KServe*
+[source]
+----
+grpcurl -proto text-generation-inference/proto/generation.proto -d '{"requests": [{"text":"<text>"}]}' -H 'Authorization: Bearer <token>' -insecure <inference_endpoint_url>:443 fmaas.GenerationService/Generate 
+----
+
+*OpenVINO Model Server*
+[source]
+----
+curl -ks <inference_endpoint_url>/v2/models/<model_name>/infer -d '{ "model_name": "<model_name>", "inputs": [{ "name": "<name_of_model_input>", "shape": [<shape>], "datatype": "<data_type>", "data": [<data>] }]}' -H 'Authorization: Bearer <token>'
+----
+
+*vLLM ServingRuntime for KServe*
+[source]
+----
+curl -v https://<inference_endpoint_url>:443/v1/chat/completions -H "Content-Type: application/json" -d '{ "messages": [{ "role": "<role>", "content": "<content>" }] -H 'Authorization: Bearer <token>'
+----
+--
+endif::[]
+
+[role='_additional-resources']
+== Additional resources
+* link:https://github.com/IBM/text-generation-inference[Text Generation Inference Server (TGIS)^]
+* link:https://caikit.readthedocs.io/en/latest/autoapi/caikit/index.html[Caikit API documentation^]
+* link:https://github.com/markstur/caikit-embeddings[Caikit Text Embedding GitHub project^]
+* link:https://docs.openvino.ai/2023.3/ovms_docs_rest_api_kfs.html[OpenVINO KServe-compatible REST API documentation^]
+* link:https://platform.openai.com/docs/api-reference/introduction[OpenAI API documentation]^
```

Comment:
Not sure if we want to directly link to openai. Maybe it's worth it to document that the api is self-documenting (using openAPI) on the `/docs` endpoint